pennylane/beta/plugins/expt_tensornet.py

```diff
@@ -19,43 +19,232 @@
 from itertools import product
 
 import numpy as np
+from numpy.linalg import eigh
 
 try:
     import tensornetwork as tn
 except ImportError as e:
     raise ImportError("expt.tensornet device requires TensorNetwork>=0.2")
 
 from pennylane._device import Device
-from pennylane.plugins.default_qubit import (
-    CNOT,
-    CSWAP,
-    CZ,
-    SWAP,
-    CRot3,
-    CRotx,
-    CRoty,
-    CRotz,
-    H,
-    Rot3,
-    Rotx,
-    Roty,
-    Rotz,
-    Rphi,
-    S,
-    T,
-    X,
-    Y,
-    Z,
-    hermitian,
-    identity,
-    Toffoli,
-    spectral_decomposition,
-    unitary,
-)
 
 # tolerance for numerical errors
 tolerance = 1e-10
 
+
+#========================================================
+#  utilities
+#========================================================
+
+def spectral_decomposition(A):
+    r"""Spectral decomposition of a Hermitian matrix.
+
+    Args:
+        A (array): Hermitian matrix
+
+    Returns:
+        (vector[float], list[array[complex]]): (a, P): eigenvalues and hermitian projectors
+            such that :math:`A = \sum_k a_k P_k`.
+    """
+    d, v = eigh(A)
+    P = []
+    for k in range(d.shape[0]):
+        temp = v[:, k]
+        P.append(np.outer(temp, temp.conj()))
+    return d, P
+
+
+#========================================================
+#  fixed gates
+#========================================================
+
+I = np.eye(2)
```

Comment:
These parts would be TODO to use the corresponding ``Operation.matrix`` throughout the ``tensornet`` and ``tensornet.tf`` plugins, correct?

Reply:
Kind of. This would work for the non-parametrized gates, but the parametrized versions will have to continue to have their own definitions (as they will need to be written using TF, not numpy).